tests/test_config.py

```diff
@@ -73,3 +90,165 @@ def test_reference_anchor_for_module(repo):
+    assert len(cleaned_and_sorted) == 3
+    for i, person in enumerate(cleaned_and_sorted):
+        assert person == file_config.config["people"][i]
+
+
+@mock.patch("mots.config.get_bmo_data")
+def test_clean_removed_user_with_refresh(
+    get_bmo_data, repo, config, test_bmo_user_data
+):
+    """Test that removing a user with a refresh updates from BMO."""
+    get_bmo_data.return_value = test_bmo_user_data
+
+    file_config = FileConfig(repo / "mots.yml")
+    file_config.load()
+
+    assert file_config.config["people"] == config["people"]
+
+    file_config.config["people"].pop()
+    file_config.write()
+
+    clean(file_config, refresh=True)
+
+    cleaned_and_sorted = sorted(file_config.config["people"], key=itemgetter("bmo_id"))
+    assert len(cleaned_and_sorted) == 3
+    for i, person in enumerate(cleaned_and_sorted):
+        assert person["name"] == test_bmo_user_data[i]["real_name"]
+        assert person["nick"] == test_bmo_user_data[i]["nick"]
+
+
+@mock.patch("mots.config.get_bmo_data")
+def test_clean_changed_names(get_bmo_data, repo, config, test_bmo_user_data):
+    """Test that updated BMO data will be reflected when cleaning."""
+    get_bmo_data.return_value = test_bmo_user_data
+
+    file_config = FileConfig(repo / "mots.yml")
+    file_config.load()
+
+    assert file_config.config["people"] == config["people"]
+
+    clean(file_config)
+
+    for person in file_config.config["people"]:
+        assert person["name"] == test_bmo_user_data[person["bmo_id"]]["real_name"]
+        assert person["nick"] == test_bmo_user_data[person["bmo_id"]]["nick"]
+
+
+@mock.patch("mots.config.get_bmo_data")
+def test_clean_changed_names_no_refresh(get_bmo_data, repo, config, test_bmo_user_data):
+    """Test that updated BMO data has no effect when cleaning without refresh."""
+    get_bmo_data.return_value = test_bmo_user_data
+
+    file_config = FileConfig(repo / "mots.yml")
+    file_config.load()
+
+    assert file_config.config["people"] == config["people"]
+
+    clean(file_config, refresh=False)
+
+    assert sorted(file_config.config["people"], key=itemgetter("bmo_id")) == sorted(
+        config["people"], key=itemgetter("bmo_id")
+    )
+
+
+@mock.patch("mots.config.get_bmo_data")
+def test_clean_added_user_no_refresh(get_bmo_data, repo, config, test_bmo_user_data):
+    """Test that updated BMO data has no effect when cleaning without refresh."""
+    get_bmo_data.return_value = test_bmo_user_data
+
+    file_config = FileConfig(repo / "mots.yml")
+    file_config.load()
+
+    assert file_config.config["people"] == config["people"]
+
+    file_config.config["people"].insert(1, {"bmo_id": 3})
+    file_config.write()
+
+    assert len(file_config.config["people"]) == 5
+
+    # Check that the new entry is unaffected after write.
+    for person in file_config.config["people"]:
+        if person["bmo_id"] == 3:
+            # Ensure only the bmo_id key is there.
+            assert len(person) == 1
```

Comment:
This would be better as `assert len(person) == 1, "Only the bmo_id key should be present."`